source/extensions/filters/http/ext_proc/http_client/http_client_impl.cc

```diff
@@ -1,20 +1,86 @@
 #include "source/extensions/filters/http/ext_proc/http_client/http_client_impl.h"
 
 #include "source/common/common/enum_to_int.h"
+#include "source/common/http/header_map_impl.h"
+#include "source/common/http/message_impl.h"
 #include "source/common/http/utility.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace HttpFilters {
 namespace ExternalProcessing {
 
+void ExtProcHttpClient::setCallbacks(RequestCallbacks* callbacks) {
+  callbacks_ = callbacks;
+  ENVOY_LOG(debug, "ext_proc http client set callbacks_ == nullptr? {}", (callbacks_ == nullptr));
+}
+
+void ExtProcHttpClient::sendRequest(envoy::service::ext_proc::v3::ProcessingRequest&& req,
+                                    const uint64_t stream_id) {
+  // Cancel any active requests.
+  cancel();
+
+  // Transcode req message into JSON string.
+  auto req_in_json = MessageUtil::getJsonStringFromMessage(req);
+  if (req_in_json.ok()) {
+    const auto http_uri = config_.http_service().http_service().http_uri();
+    const std::string cluster = http_uri.cluster();
+    const std::string uri = http_uri.uri();
+    absl::string_view host, path;
+    Envoy::Http::Utility::extractHostPathFromUri(uri, host, path);
+    ENVOY_LOG(debug, " Ext_Proc HTTP client send request to cluster {}, uri {}, host {}, path {}",
+              cluster, uri, host, path);
+
+    const auto thread_local_cluster = context().clusterManager().getThreadLocalCluster(cluster);
```

Comment:
Please add nullptr check for `thread_local_cluster ` If can be nullptr if cluster is not 
It is important  to avoid data plane crash

Reply:
done